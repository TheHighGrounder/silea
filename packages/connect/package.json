{
	"name": "@sileajs/connect",
	"version": "0.0.2",
	"description": "The state manager of Silea. Similar to MobX but much more lightweight because it's based on ES6 proxies.",
	"keywords": [
		"silea",
		"connect",
		"state-manager"
	],
	"homepage": "https://sileajs.com",
	"license": "SEE LICENSE IN LICENSE",
	"main": "src/index.js",
	"types": "src/index.d.ts",
	"repository": {
		"type": "git",
		"url": "https://github.com/SileaJS/silea.git"
	},
	"scripts": {
		"test:ci": "../../node_modules/.bin/jest --ci --coverage",
		"test": "../../node_modules/.bin/jest --watch",
		"test:inspect": "node --inspect ../../node_modules/.bin/jest --watch --no-cache --runInBand"
	},
	"dependencies": {
		"@sileajs/error": "0.0.2"
	},
	"devDependencies": {
		"react-test-renderer": "16.13.1"
	},
	"publishConfig": {
		"access": "public"
	}
}
