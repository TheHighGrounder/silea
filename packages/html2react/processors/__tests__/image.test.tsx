import React from "react";
import { render } from "@testing-library/react";
import { createStore } from "@sileajs/connect";
import { Html2React } from "../../src/libraries/component";
import image from "../image";
import Html2ReactPackage from "../../src";

const store = createStore(Html2ReactPackage);
store.libraries.html2react.processors.push(image);

jest.mock("@sileajs/components/image", () => "mocked-img");

describe("Image Processor", () => {
	it("should process images from src", () => {
		const { container } = render(
			<Html2React
				html="<img src='https://sileajs.com/logo.png' alt='Silea' width='500' />"
				{...store}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it("should process images from data-src", () => {
		const { container } = render(
			<Html2React
				html="<img data-src='https://sileajs.com/logo.png' alt='Silea' width='500' />"
				{...store}
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
