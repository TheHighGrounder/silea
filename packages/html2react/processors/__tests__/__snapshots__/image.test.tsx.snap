// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Processor should process images from data-src 1`] = `
<div>
  <mocked-img
    alt="Silea"
    data-src="https://sileajs.com/logo.png"
    src="https://sileajs.com/logo.png"
    width="500"
  />
</div>
`;

exports[`Image Processor should process images from src 1`] = `
<div>
  <mocked-img
    alt="Silea"
    src="https://sileajs.com/logo.png"
    width="500"
  />
</div>
`;
